packages/itwin/ec3-widget/src/components/EC3/EC3Config.ts

```diff
@@ -40,6 +41,26 @@ export interface EC3ConfigCommonProps {
    * The base path for the Carbon Calculation API endpoints. If not specified, it defaults to CARBON_CALCULATION_BASE_PATH from @itwin/insights-client.
    */
   carbonCalculationBasePath?: string;
+
+  /**
+   * A custom implementation of ReportsClient.
+   */
+  reportsClient?: IReportsClient;
+
+  /**
+   * A custom implementation of OdataClient.
+   */
+  oDataClient?: IOdataClient;
+
+  /**
+   * A custom implementation of EC3JobsClient.
+   */
+  ec3JobsClient?: IEC3JobsClient;
+
+  /**
+   * A custom implementation of EC3ConfigurationClient.
```

Comment:
It would be good to document that if client is specified, the base matching base path is ignored.

Reply:
Done.